docs/content/en/docs/tasks/troubleshoot/troubleshooting.md

```diff
@@ -138,20 +125,148 @@ If you don't have any other docker containers running you may want to run `docke
 You may want to restart Docker.
 To restart Docker on Ubuntu `sudo systemctl restart docker`.
 
-### Issues detected with selected template
-```
-Issues detected with selected template. Details: - -1:-1:VALUE_ILLEGAL: No supported hardware versions among [vmx-15]; supported: [vmx-04, vmx-07, vmx-08, vmx-09, vmx-10, vmx-11, vmx-12, vmx-13].
-```
-Our upstream dependency on CAPV makes it a requirement that you use vSphere 6.7 update 3 or newer.
-Make sure your ESXi hosts are also up to date.
-
 ### Waiting for cert-manager to be available... Error: timed out waiting for the condition
 ```
 Failed to create cluster {"error": "error initializing capi resources in cluster: error executing init: Fetching providers\nInstalling cert-manager Version=\"v1.1.0\"\nWaiting for cert-manager to be available...\nError: timed out waiting for the condition\n"}
 ```
 This is likely a [Memory or disk resource problem]({{< relref "#memory-or-disk-resource-problem" >}}).
 You can also try using techniques from [Generic cluster unavailable]({{< relref "#generic-cluster-unavailable" >}}).
 
+
+### The connection to the server localhost:8080 was refused 
+```
+Performing provider setup and validations
+Creating new bootstrap cluster
+Installing cluster-api providers on bootstrap cluster
+Error initializing capi in bootstrap cluster	{"error": "error waiting for capi-kubeadm-control-plane-controller-manager in namespace capi-kubeadm-control-plane-system: error executing wait: The connection to the server localhost:8080 was refused - did you specify the right host or port?\n"}
+Failed to create cluster	{"error": "error waiting for capi-kubeadm-control-plane-controller-manager in namespace capi-kubeadm-control-plane-system: error executing wait: The connection to the server localhost:8080 was refused - did you specify the right host or port?\n"}
+```
+This is likely a [Memory or disk resource problem]({{< relref "#memory-or-disk-resource-problem" >}}).
+
+### Generic cluster unavailable
+Troubleshoot more by inspecting bootstrap cluster or workload cluster (depending on the stage of failure) using kubectl commands. 
+```
+kubectl get pods -A --kubeconfig=<kubeconfig>
+kubectl get nodes -A --kubeconfig=<kubeconfig>
+kubectl get logs <podname> -n <namespace> --kubeconfig=<kubeconfig>
+....
+```
+Capv troubleshooting guide: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/master/docs/troubleshooting.md#debugging-issues
+
+## Bare Metal troubleshooting
+
+### Bootstrap cluster fails to come up
+
+```
+Error: creating bootstrap cluster: executing create cluster: ERROR: failed to create cluster: node(s) already exist for a cluster with the name \"cluster-name\"
+, try rerunning with —force-cleanup to force delete previously created bootstrap cluster
+```
+
+The cluster already exists, so you can delete the old one:
+
+```bash
+kind delete cluster --name cluster-name
+```
```

Comment:
```suggestion
```
Error: creating bootstrap cluster: executing create cluster: ERROR: failed to create cluster: node(s) already exist for a cluster with the name \"cluster-name\"
, try rerunning with —force-cleanup to force delete previously created bootstrap cluster
```

If you observe the above cluster creation error and have already attempted to create the cluster using the recommended `--force-cleanup` flag, you can manually clean the existing cluster with the following command.

```bash
kind delete cluster --name <cluster-name>
```
```